gh-badges/lib/badge-renderers.js

```diff
@@ -0,0 +1,596 @@
+
+      <g fill="#fff" text-anchor="middle" ${this.constructor.fontFamily} font-size="110">
+        ${this.renderedLogo}
+        ${this.renderedLabel}
+        ${this.renderedMessage}
+      </g>`
+    )
+  }
+}
+
+class Flat extends Badge {
+  static get fontFamily() {
+    return fontFamily
+  }
+
+  static get height() {
+    return 20
+  }
+
+  static get verticalMargin() {
+    return 0
+  }
+
+  static get shadow() {
+    return true
+  }
+
+  render() {
+    return renderBadge(
+      {
+        links: this.links,
+        leftWidth: this.leftWidth,
+        rightWidth: this.rightWidth,
+        height: this.constructor.height,
+      },
+      `
+      <linearGradient id="s" x2="0" y2="100%">
+        <stop offset="0" stop-color="#bbb" stop-opacity=".1"/>
+        <stop offset="1" stop-opacity=".1"/>
+      </linearGradient>
+
+      <clipPath id="r">
+        <rect width="${this.width}" height="${this.constructor.height}" rx="3" fill="#fff"/>
+      </clipPath>
+
+      <g clip-path="url(#r)">
+        <rect width="${this.leftWidth}" height="${this.constructor.height}" fill="${this.labelColor}"/>
+        <rect x="${this.leftWidth}" width="${this.rightWidth}" height="${this.constructor.height}" fill="${this.color}"/>
+        <rect width="${this.width}" height="${this.constructor.height}" fill="url(#s)"/>
+      </g>
+
+      <g fill="#fff" text-anchor="middle" ${this.constructor.fontFamily} font-size="110">
+        ${this.renderedLogo}
+        ${this.renderedLabel}
+        ${this.renderedMessage}
+      </g>`
+    )
+  }
+}
+
+class FlatSquare extends Badge {
+  static get fontFamily() {
+    return fontFamily
+  }
+
+  static get height() {
+    return 20
+  }
+
+  static get verticalMargin() {
+    return 0
+  }
+
+  static get shadow() {
+    return false
+  }
+
+  render() {
+    return renderBadge(
+      {
+        links: this.links,
+        leftWidth: this.leftWidth,
+        rightWidth: this.rightWidth,
+        height: this.constructor.height,
+      },
+      `
+      <g shape-rendering="crispEdges">
+        <rect width="${this.leftWidth}" height="${this.constructor.height}" fill="${this.labelColor}"/>
+        <rect x="${this.leftWidth}" width="${this.rightWidth}" height="${this.constructor.height}" fill="${this.color}"/>
+      </g>
+
+      <g fill="#fff" text-anchor="middle" ${this.constructor.fontFamily} font-size="110">
+        ${this.renderedLogo}
+        ${this.renderedLabel}
+        ${this.renderedMessage}
+      </g>`
+    )
+  }
+}
+
+function plastic(params) {
+  const badge = new Plastic(params)
+  if (params.minify) {
+    return stripXmlWhitespace(badge.render())
+  }
+  return badge.render()
+}
+
+function flat(params) {
+  const badge = new Flat(params)
+  if (params.minify) {
+    return stripXmlWhitespace(badge.render())
+  }
+  return badge.render()
+}
+
+function flatSquare(params) {
+  const badge = new FlatSquare(params)
+  if (params.minify) {
+    return stripXmlWhitespace(badge.render())
```

Comment:
Not necessary for merging this, though it would be nice to reduce the duplication of the minify logic. Perhaps it could go in the Badge superclass or in the calling code.